packages/components/src/components/hds/link/types.ts

```diff
@@ -0,0 +1,39 @@
+import type { HdsInteractiveArgs } from '../interactive';
+
+export enum HdsLinkIconPositionValues {
+  Leading = 'leading',
+  Trailing = 'trailing',
+}
+
+export type HdsLinkIconPositions = `${HdsLinkIconPositionValues}`;
+
+export enum HdsLinkColorValues {
+  Primary = 'primary',
+  Secondary = 'secondary',
+}
+
+export type HdsLinkColors = `${HdsLinkColorValues}`;
+
+export enum HdsLinkStandaloneSizeValues {
+  Small = 'small',
+  Medium = 'medium',
+  Large = 'large',
+}
+
+export type HdsLinkStandaloneSizes = `${HdsLinkStandaloneSizeValues}`;
```

Comment:
still need to update this to follow the new rules we set.

Reply:
Update: [we found that switching to enum unions caused type issues w/ glint](https://hashicorp.slack.com/archives/C06NMSYDZPA/p1710852346759879) and have decided to proceed with the template literal approach for now.